### DISCUSSION TITLE
Unify .md & .mdx implementations

### DISCUSSION THREAD

**Discussion by jamespohalloran** [tinacms/tinacms#2869] [Ideas]
Currently in our docs, we have separate approaches for implementing [markdown](https://tina.io/docs/editing/markdown/) or [mdx](https://tina.io/docs/editing/mdx/) content.

For the Markdown approach, implementation is as follows:
- the content type on the body field is "string"
- You need to install a separate package `react-tinacms-editor`
- You need to add the cms plugin `cms.plugins.add(field.MarkdownFieldPlugin)`
- The markdown is rendered on the page with the `React-Markdown` (or somthing similar)

For the MDX approach:
- the content type on the body field type is set to "rich-text"
- the content is rendered on the page with the `TinaMarkdown` component

The `react-tinacms-editor` editor also uses prosemirror, while the default rich-text editor uses slate.

## Problems:
- If a user decides that they want to allow the editor to add templates, migrating from .md to .mdx is a bit of a pain. 
- We have to maintain 2 separate editors (slate & prosemirror), 2 different docs, etc. 

## Should we unify the `.md` & `.mdx` approaches?

Should we deprecate the markdown approach, and instead encourage users use `rich-text` for the body on all `body` fields?

- We'd have to make it clear that if your format was .md, `templates` wouldn't be supported (I assume). 
- To render your markdown content, you'd need to use our `TinaMarkdown` component as-is. Could we provide a utility for getting the raw markdown string from the response so that a user could use their existing React-Markdown plugin?

**Comment by arunavo4** [comment]
@jamespohalloran I am currently facing this issue, How to get the raw string from the rich-text to use my own markdown plugin I dont want to use the Tina-Markdown Component.

**Comment by zweinz** [comment]
Similarly, I would love to be able to mix .md and .mdx files, since the .md files are edited with an external editor (Ulysses) that can't handle .mdx extensions.

**Comment by blu3gui7ar** [comment]
```
import { stringifyMDX } from '@tinacms/mdx'

const field = { type: "rich-text", name: "body", label: "Body", isBody: true }
const md = stringifyMDX(<tinacms-syntax-tree>, field, (url) => url)
```

This seems to generate the original markdown.
